subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+        #            "flg_liberado": "0",
+        #            "dsc_data_liberacao": null,
+        #            "dsc_data": "2011-06-12T21:06:42",
+        #            "dsc_metaphone_us": "BRKNKBT0SSN",
+        #            "dsc_metaphone_br": "BRKNKBTTMPRT",
+        #            "episodios": null,
+        #            "flg_seriado": null,
+        #            "last_used": "1372569074",
+        #            "deleted": false
+        #        },
+        #        "sort": [
+        #            "4"
+        #        ]
+        #    }
+        # ]
+        #
+        # Notes:
+        #  tipo: Defines if the entry is a movie or a tv show (or a collection??)
+        #  imdb_id: Sometimes it appears as a number and sometimes as a string prefixed with tt
+        #  temporada: Sometimes is ``null`` and season information should be extracted from dsc_nome_br
+
+        results = json.loads(r.text)
+
+        # type, title, series, season, year follow guessit properties names
+        mapping = dict(
+            id='id_filme',
+            type='tipo',
+            title='dsc_nome',
+            series='dsc_nome',
+            season='temporada',
+            year='dsc_data_lancamento',
+            title_br='dsc_nome_br',
+            imdb_id='id_imdb'
+        )
+
+        # movie and episode values follow guessit type values
+        type_map = {
+            'M': 'movie',
+            'S': 'episode',
+            'C': 'episode'  # Considering C as episode. Probably C stands for Collections
+        }
+
+        # Regex to extract the season number. e.g.: 3\u00aa Temporada, 1a Temporada, 2nd Season
+        season_re = re.compile('.*? - (\d{1,2}).*?((emporada)|(Season))', re.IGNORECASE)
+
+        # Regex to extract the IMDB id. e.g.: tt02342
+        imdb_re = re.compile('t{0,2}(\d+)')
+
+        candidates = []
+        for result in results:
+            entry = result['_source']
+            item = {k: entry.get(v) for k, v in mapping.items()}
+            item['type'] = type_map.get(item.get('type'), 'movie')
+            item['imdb_id'] = (lambda m: m.group(1) if m else None)(imdb_re.search(item.get('imdb_id')))
+
+            # Season information might be missing and it should be extracted from 'title_br'
+            if not item.get('season') and item.get('title_br'):
+                item['season'] = (lambda m: m.group(1) if m else None)(season_re.search(item.get('title_br')))
+
+            # Some string fields are actually integers
+            for field in ['season', 'year', 'imdb_id']:
+                item[field] = (lambda v: int(v) if v and v.isdigit() else None)(item.get(field))
+
+            # ignoring episode match since this first step is only about movie/season information
+            if self.matches(params, item, ignore_episode=True):
+                candidates.append(dict(item))
+
+        logger.debug('Titles found: %s', candidates)
+        return candidates
+
+    def query(self, language, video, params):
```

Comment:
Same here, explicit args rather than params would be more readable and a cleaner API. The goal of the  `query` method is to be as close as possible to a query to the provider with minimal filtering (as this will occur in the scoring part anyway)